ebssurrogate/scripts/surrogate-bootstrap.sh

```diff
@@ -0,0 +1,267 @@
+	docker run -itd --name ccachedata "${DOCKER_IMAGE}:${DOCKER_IMAGE_TAG}" sh
+	docker exec -itd ccachedata mkdir -p /build/ccache
+}
+
+# Create fstab
+function create_fstab {
+	FMT="%-42s %-11s %-5s %-17s %-5s %s"
+cat > "/mnt/etc/fstab" << EOF
+$(printf "${FMT}" "# DEVICE UUID" "MOUNTPOINT" "TYPE" "OPTIONS" "DUMP" "FSCK")
+$(findmnt -no SOURCE /mnt | xargs blkid -o export | awk -v FMT="${FMT}" '/^UUID=/ { printf(FMT, $0, "/", "ext4", "defaults,discard", "0", "1" ) }')
+$(findmnt -no SOURCE /mnt/boot/efi | xargs blkid -o export | awk -v FMT="${FMT}" '/^UUID=/ { printf(FMT, $0, "/boot/efi", "vfat", "umask=0077", "0", "1" ) }')
+EOF
+	unset FMT
+}
+
+function setup_chroot_environment {
+	# Bootstrap Ubuntu into /mnt
+	debootstrap --arch ${ARCH} --variant=minbase focal /mnt
+
+	# Update ec2-region
+	REGION=$(curl --silent --fail http://169.254.169.254/latest/meta-data/placement/availability-zone | sed -E 's|[a-z]+$||g')
+	sed -i "s/REGION/${REGION}/g" /tmp/sources.list
+	cp /tmp/sources.list /mnt/etc/apt/sources.list
+
+	if [ "${ARCH}" = "arm64" ]; then
+		create_fstab
+	fi
+
+	# Create mount points and mount the filesystem
+	mkdir -p /mnt/{dev,proc,sys}
+	mount --rbind /dev /mnt/dev
+	mount --rbind /proc /mnt/proc
+	mount --rbind /sys /mnt/sys
+
+        # Create build mount point and mount 
+	mkdir -p /mnt/tmp
+	mount /dev/xvdc /mnt/tmp
+	chmod 777 /mnt/tmp
+
+	# Copy the bootstrap script into place and execute inside chroot
+	cp /tmp/chroot-bootstrap.sh /mnt/tmp/chroot-bootstrap.sh
+	chroot /mnt /tmp/chroot-bootstrap.sh
+	rm -f /mnt/tmp/chroot-bootstrap.sh
+
+	# Copy the nvme identification script into /sbin inside the chroot
+	mkdir -p /mnt/sbin
+	cp /tmp/ebsnvme-id /mnt/sbin/ebsnvme-id
+	chmod +x /mnt/sbin/ebsnvme-id
+
+	# Copy the udev rules for identifying nvme devices into the chroot
+	mkdir -p /mnt/etc/udev/rules.d
+	cp /tmp/70-ec2-nvme-devices.rules \
+		/mnt/etc/udev/rules.d/70-ec2-nvme-devices.rules
+
+	#Copy custom cloud-init
+	rm -f /mnt/etc/cloud/cloud.cfg
+	cp /tmp/cloud.cfg /mnt/etc/cloud/cloud.cfg
+
+	sleep 2
+}
+
+function download_ccache {
+	docker cp ccachedata:/build/ccache/. /mnt/tmp/ccache
+}
+
+function execute_playbook {
+	# Run Ansible playbook
+	#export ANSIBLE_LOG_PATH=/tmp/ansible.log && export ANSIBLE_DEBUG=True && export ANSIBLE_REMOTE_TEMP=/mnt/tmp 
+	export ANSIBLE_LOG_PATH=/tmp/ansible.log && export ANSIBLE_REMOTE_TEMP=/mnt/tmp 
+	ansible-playbook -c chroot -i '/mnt,' /tmp/ansible-playbook/ansible/playbook.yml --extra-vars " $ARGS"
+}
+
+function update_systemd_services {
+	# Disable vector service and set timer unit.
+	cp -v /tmp/vector.timer /mnt/etc/systemd/system/vector.timer
+	rm -f /mnt/etc/systemd/system/multi-user.target.wants/vector.service
+	ln -s /etc/systemd/system/vector.timer /mnt/etc/systemd/system/multi-user.target.wants/vector.timer
+
+	# Uncomment below to Disable postgresql service during first boot.
+	# rm -f /mnt/etc/systemd/system/multi-user.target.wants/postgresql.service
```

Comment:
was this meant to be uncommented?

Reply:
yes, right. Updated PR and tested the changes.